langserve/serialization.py

```diff
@@ -90,6 +93,16 @@ class _LangChainEncoder(json.JSONEncoder):
     def default(self, obj) -> Any:
         if isinstance(obj, BaseModel):
             return obj.dict()
+        elif isinstance(obj, UUID):
```

Comment:
can we just do this with `orjson`, or does that make the deserialization hard?

Reply:
orjson doesn't support object hooks by design, so we can't use just orjson, we'll still need to use our own decoder to allow users to hook into the encoder.

Hooking into the encoder seems like a must for a user to be able to add information for type discrimination on unions

Reply:
I think it does support? https://github.com/ijl/orjson#default

Reply:
Ah you're right -- it looks like encoding path supports, but not decoding path